Source/WebInspectorUI/UserInterface/Views/Popover.js

```diff
@@ -212,8 +212,12 @@ WI.Popover = class Popover extends WI.Object
             WI.Popover._canvasContext = new WeakRef(context);
         }
 
-        context.canvas.width = width;
-        context.canvas.height = height;
+        let canvas = context.canvas;
+        canvas.style.width = width + "px";
+        canvas.style.height = height + "px";
```

Comment:
is this actually necessary?  i would think that the `inset: 0` would ensure that the size of the `<canvas>` never becomes larger than desired 🤔

Reply:
This is a key part of the change that actually fixes the issue. In order for the canvas to have the number of actual pixels we want on Retina displays its width/height attribute have to be the number of actual pixels, otherwise the drawing results are blurry. But, as this bug demonstrates, the canvas will then be sized to display at that number of points on the screen, and not that number of pixels, resulting in the double-sized popover on Retina displays unless we explicitly set a CSS width+height for layout purposes. This used to all be handled automagically before the regression point because the size of the canvas for layout would always match the size of the element the canvas was for, no matter what width/height we used to create the underlying context for drawing.

Reply:
hmm i guess my question is "why doesn't the `inset: 0` do what we want?"

would specifying `width: 100%; height: 100%;` in the CSS alongside the `inset: 0` work instead?

Reply:
Oh lol, that sure does work. And makes the entire rest of the change irrelevant 😅